### ISSUE TITLE
`automagic` command

### ISSUE THREAD

**Issue by YamatoSecurity** [Yamato-Security/takajo#132]
`automagic: automatically executes as many commands as possible and output results to a new folder`

options:
```
-l, --level [optional] filter by minimal level of alerts (default: informational)
-t, --timeline [required]
-o, --output <folder name> [required]
-q, --quiet: do not display the launch banner (default: false)
```

Example: `automagic -t ../hayabusa/timeline.jsonl [--level low] -o case-1`

Note: if the folder exists, display an error message saying `This folder <folder name> exists. Please specify a new folder name.`

This command will execute the following commands (if possible in parallel) in the background:
```
extract-scriptblocks -t ../hayabusa/timeline.jsonl -l <level> -o case-1/scriptblock-logs/
list-domains -t ../hayabusa/timeline.jsonl -o case-1/ListDomains.txt
list-domains -t ../hayabusa/timeline.jsonl -s -w -o case-1/ListDomains-Detailed.txt
list-hashes -t ../hayabusa/case-1.jsonl -l <level> -o case-1/hashes/
list-ip-addresses -t ../hayabusa/timeline.jsonl -o case-1/IP-Addresses.txt
stack-cmdlines -t ../hayabusa/timeline.jsonl --level <leve> -o case-1/cmdlines.csv
stack-computers -t ../hayabusa/timeline.jsonl --level <level> -o case-1/TargetComputers.csv
stack-computers -t ../hayabusa/timeline.jsonl --level <level> --sourceComputers -o case-1/SourceComputers.csv
stack-dns -t ../hayabusa/timeline.jsonl --level <level>  -o case-1/DNS.csv
stack-ip-addresses -t ../hayabusa/timeline.jsonl --level <level> -o case-1/SourceIP-Addresses.csv
stack-ip-addresses -t ../hayabusa/timeline.jsonl --level <level> --targetIpAddresses -o case-1/TargetIP-Addresses.csv
stack-logons -t ../hayabusa/timeline.jsonl -o case-1/Logons.csv
stack-processes -t ../hayabusa/timeline.jsonl --level <level> -o case-1/Processes.csv
stack-services -t ../hayabusa/timeline.jsonl --level <level> -o case-1/Services.csv
stack-tasks -t ../hayabusa/timeline.jsonl --level <level> -o case-1/Tasks.csv
stack-users -t ../hayabusa/timeline.jsonl --level <level> -o case-1/TargetUsers.csv
stack-users -t ../hayabusa/timeline.jsonl --level <level> --filterSystemAccounts=false --filterComputerAccounts=false -o case-1/TargetUsers-NoFiltering.csv
stack-users -t ../hayabusa/timeline.jsonl --level <level> --sourceUsers -o users.csv
stack-users -t ../hayabusa/timeline.jsonl --level <level> --sourceUsers --filterSystemAccounts=false --filterComputerAccounts=false -o case-1/SourceUsers-NoFiltering.csv
timeline-logon -t ../hayabusa/timeline.jsonl -o case-1/LogonTimeline.csv
timeline-partition-diagnostic -t ../hayabusa/timeline.jsonl -o case-1/PartitionDiagnosticTimeline.csv
timeline-suspicious-processes -t ../hayabusa/timeline.jsonl --level <level> -o case-1/SuspiciousProcesses.csv
timeline-tasks -t ../hayabusa/timeline.jsonl -o case-1/TaskTimeline.csv
ttp-summary -t ../hayabusa/timeline.jsonl -o case-1/TTP-Summary.csv
ttp-visualize -t ../hayabusa/timeline.jsonl -o case-1/MitreTTP-Heatmap.json
```

@fukusuket What do you think about a command that automates all of these?

**Comment by fukusuket** [comment]
@YamatoSecurity 
Sounds good! I'll try implementing it💪 We need to refactor first, so I'll start by thinking about the structure of the code :)

**Comment by fukusuket** [comment]
@YamatoSecurity @hitenkoku 
[Along with the refactoring](https://github.com/Yamato-Security/takajo/pull/135), I created a protocol for `automagic`!
I will create a separate PR for the `automagic`, but if you have any requests, I would appreciate it if you could let me know.

You can use the `automagic` command by commenting out the line below.
https://github.com/Yamato-Security/takajo/blob/refactor-for-automagic-command/src/takajo.nim#L110-L121

In the prototype above, the standard output is too noisy, so I would like to improve it ...
